src/program/latex/mod.rs

```diff
@@ -0,0 +1,695 @@
+//! LaTeX circuit generation for quil programs.
+//!
+//! This module enables generating quantum circuits using the LaTeX subpackage
+//! TikZ/[`Quantikz`] for a given quil [`Program`]. This feature is callable on
+//! [`Program`] and returns a LaTeX string which can be rendered in a LaTeX
+//! visualization tool. Be aware that not all Programs can be serialized as
+//! LaTeX. If a [`Program`] contains a gate or modifier not mentioned in the
+//! [Supported Gates and Modifiers](#supported-gates-and-modifiers) section
+//! below, an error will be returned detailing which instruction or gate is
+//! unsupported in the Program being processed.
+//!
+//! # Supported Gates and Modifiers
+//!
+//!   - Pauli Gates:           `I`, `X`, `Y`, `Z`
+//!   - Hadamard Gate:         `H`
+//!   - Phase Gate:            `PHASE`, `S`, `T`
+//!   - Controlled Phase Gate: `CZ`, `CPHASE`
+//!   - Controlled X Gates:    `CNOT`, `CCNOT`
+//!   - User-Defined Gates:    `DEFGATE`
+//!   - Modifiers:             `CONTROLLED`, `DAGGER`
+//!
+//! [`Quantikz`]: https://arxiv.org/pdf/1809.03842.pdf
+
+mod diagram;
+
+use std::{collections::HashSet, fmt};
+
+use crate::instruction::{Instruction, Qubit};
+use crate::Program;
+
+use self::settings::RenderSettings;
+use diagram::{wire::Wire, Diagram};
+
+pub mod settings;
+
+/// The structure of a LaTeX document. Typically a LaTeX document contains
+/// metadata defining the setup and packages used in a document within a header
+/// and footer while the body contains content and controls its presentation.
+struct Document {
+    header: String,
+    body: String,
+    footer: String,
+}
+
+impl Default for Document {
+    fn default() -> Self {
+        Self {
+            header: r"\documentclass[convert={density=300,outext=.png}]{standalone}
+\usepackage[margin=1in]{geometry}
+\usepackage{tikz}
+\usetikzlibrary{quantikz}
+\begin{document}
+\begin{tikzcd}"
+                .to_string(),
+            body: String::new(),
+            footer: r"\end{tikzcd}
+\end{document}"
+                .to_string(),
+        }
+    }
+}
+
+impl fmt::Display for Document {
+    /// Returns the entire document in LaTeX string.
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}{}", self.header, self.body, self.footer)
+    }
+}
+
+/// Available commands used for building circuits with the same names taken
+/// from the ``Quantikz`` documentation for easy reference. LaTeX string denoted
+/// inside `backticks`.
+#[derive(Clone, Debug, derive_more::Display, PartialEq, Eq, Hash)]
+pub(crate) enum RenderCommand {
+    /// Make a qubit "stick out" from the left.
+    #[display(fmt = "\\lstick{{\\ket{{q_{{{_0}}}}}}}")]
+    Lstick(u64),
```

Comment:
Design note: if the principle of `RenderCommand` is that it should faithfully and literally represent the underlying `Quantikz` syntax, then `LStick(u64)` diverges from that, because `\lstick{}` can contain any other Quantikz elements.

This approach is ok too - as a rendering helper - but I would suggest making it more intuitive in that case. ie, `Lstick` -> `LeftWireLabel`. 

`Lstick(u64)` straddles two layers of abstraction.

Reply:
Updated to `LeftWireLabel` [Change Lstick to LeftWireLabel.](https://github.com/rigetti/quil-rs/pull/145/commits/11191e08839aff4430a916dada22b4743e06b648)